ste/mgr-JobMgr.go

```diff
@@ -167,6 +168,8 @@ type jobMgr struct {
 	cancel               context.CancelFunc
 	pipelineNetworkStats *pipelineNetworkStats
 
+	exclusiveDestinationMapHolder *atomic.Value
```

Comment:
>exclusiveDestinationMapHolder *atomic.Value [](start = 1, length = 43)

I don't entirely understand the usage of sync/atomic here. We already have a mutex in place inside the exclusive string map to prevent multiple routines messing with the map at the same time, which should in theory prevent the data race condition. What special use does an *atomic.Value hold here?

Reply:
Note that I (generally) understand the usage of this package, and that I'm only really questioning this specific use of it.

---
In reply to: [331610387](https://github.com/Azure/azure-storage-azcopy/pull/638#discussion_r331610387) [](ancestors = 331610387)

Reply:
I thought there were multiple goroutines that accessed this field - i.e. that needed to get the pointer to the exclusive string map.  So I wanted the getting of the pointer to be threadsafe. (As you say, once you have the pointer, using it to access the map is fine).

I've realized just now that all of the calls to AddJobPart probably happen on the same goroutine (the one doing the enumeration) and so what I did here was unnecessary (if we are willing to rely on that assumption), since is only at AddJobPart time that we mess with it.  So I probably didn't need this.  But I don't have time to remove it at the moment.

Reply:
Ah! I didn't think there'd be a case where getting the pointer to it _wouldn't_ be thread-safe, aside from the initial setting, I guess.